rules/parser.y

```diff
@@ -235,6 +237,28 @@ extra_labels_opts  :
                      { $$ = true }
                    ;
 
+vector_matching    :
+                     { $$ = nil }
+                   | GROUP_OP '(' label_list ')'
+                     { 
+                       var err error
+                       $$, err = newVectorMatching("", $3, nil)
+                       if err != nil { yylex.Error(err.Error()); return 1 }
+                     }
+                   | GROUP_MOD GROUP_OP '(' label_list ')'
```

Comment:
I'm not sure that GROUP_MOD should be allowed without the label_list being supplied. It could result in unexpected labels making it through, and in general you want to be able to tell what labels will result from an expression just by looking at the expression - not the input data.